src/scripts/temporaryBattle/TemporaryBattleList.ts

```diff
@@ -414,6 +414,20 @@ TemporaryBattleList['Suicune 2'] = new TemporaryBattle(
         hideTrainer: true,
     }
 );
+TemporaryBattleList['Red Gyarados'] = new TemporaryBattle(
+    'Red Gyarados',
+    [new GymPokemon('Gyarados', 1200000, 40, undefined, true)],
+    undefined,
+    [new QuestLineStartedRequirement('Team Rocket Again')],
+    [new QuestLineStepCompletedRequirement('Team Rocket Again', 0)],
```

Comment:
No need for a defeated requirement, when you only need to defeat it once. So just set `[new QuestLineStepCompletedRequirement('Team Rocket Again', 0)],` to `undefined,`